examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,491 @@
+==================================================================
+
+.. contents::
+   :local:
+   :depth: 1
+
+Linear models describe situations in which the target value is expected to be
+a linear combination of the features (see the :ref:`linear_model` User Guide
+section for a description of a set of linear model methods available in
+scikit-learn).
+Coefficients in multiple linear models represent the relationship between the
+given feature (`X[i]`) and the target (`y`) assuming that all the other
+features remain constant (`conditional dependence
+<https://en.wikipedia.org/wiki/Conditional_dependence>`_).
+This is not the same thing than plotting `X[i]` versus `y` and fitting a linear
+relationship: in that case all possible values of the other features are
+added to the estimation (marginal dependence).
+
+This example will provide some hints in interpreting coefficient in linear
+models, pointing at problems that arise when either the linear model is not
+appropriate to describe the dataset, or features are correlated.
+
+We will use data from the "Current Population Survey" from 1985 to predict
+wage as a function of various features such as experience, age, or education.
+
+A description of the dataset follows.
+"""
+
+print(__doc__)
+
+import numpy as np
+import scipy as sp
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+#############################################################################
+# The dataset
+# -----------
+#
+# We fetch the data from `OpenML <http://openml.org/>`_.
+# Note that setting the parameter `as_frame` to True will retrieve the data
+# as a pandas dataframe.
+
+from sklearn.datasets import fetch_openml
+
+survey = fetch_openml(data_id=534, as_frame=True)
+
+##############################################################################
+# Then, we identify features `X` and targets `y`: the column WAGE is our
+# target variable (i.e., the variable which we want to predict).
+#
+X = survey.data[survey.feature_names]
+X.describe(include="all")
+
+##############################################################################
+# Notice that the dataset contains categorical and numerical variables.
+# Some of the categorical variables are binary variables.
+# About the numerical ones we can observe that AGE and EXPERIENCE have similar
+# distributions while the EDUCATION distribution is narrower.
+# This will give us directions on how to preprocess the data thereafter.
+
+X.head()
+
+##############################################################################
+y = survey.target.values.ravel()
+survey.target.head()
+
+###############################################################################
+# We split the sample in a train and a test dataset.
+# Only the train dataset will be used in the following exploratory analysis.
+# This is a way to emulate a real situation where predictions are performed on
+# an unknown target.
```

Comment:
Maybe clarify that the reason for splitting the data is that we don't want our analysis and decisions to be biased by our knowledge of the test data.